### PR TITLE
utils_net.py: departed dump_traceview_log and get_MSIs_and_queues_windows in this script

### PR DESCRIPTION
**Pull request by kimi1978wy**
It only need dump_traceview_log for some scripts.

Signed-off-by: Yu Wang <wyu@redhat.com>

id: 1718715

### CODE REVIEW

File: virttest/utils_net.py [hunk 1]
```diff
@@ -3997,4 +3997,20 @@ def get_MSIs_and_queues_windows(params, vm, timeout=360):
         raise exceptions.TestError("Cann't read dumped file %s: %s" %
                                    (dump_file, output))
     session_serial.close()
+    return output
+
+
+def get_MSIs_and_queues_windows(params, vm, timeout=360):
+    """
+    Get msi&queues infomation of currentwindows guest.
```
Comment (xutian) [inline virttest/utils_net.py]: blank space  need between two words

Comment (xutian) [reply virttest/utils_net.py]: sed 's/msi/MSI/g'

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

File: virttest/utils_net.py [hunk 2]
```diff
@@ -3936,19 +3924,19 @@ def _check_file_size_unchanged():
     session.cmd(kill_cmd)
 
 
-def get_MSIs_and_queues_windows(params, vm, timeout=360):
+def dump_traceview_log(params, vm, timeout=360):
```
Comment (vivianQizhu) [inline virttest/utils_net.py]: Suggest add `windows` as suffix

**Comment by PaulYuuu** [comment]
Please rebase to the current code, others LGTM

**Comment by kimi1978wy** [comment]
@vivianQizhu updated

 (1/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.netkvm_cpu_mapping.q35: PASS (155.62 s)
 (2/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.mq_enabled_chk.q35: PASS (370.24 s)

**Comment by kimi1978wy** [comment]
@xutian Updated, could you help to review this patch again?

File: virttest/utils_net.py [hunk 3]
```diff
@@ -3997,4 +3997,20 @@ def get_MSIs_and_queues_windows(params, vm, timeout=360):
         raise exceptions.TestError("Cann't read dumped file %s: %s" %
                                    (dump_file, output))
     session_serial.close()
+    return output
+
+
+def get_MSIs_and_queues_windows(params, vm, timeout=360):
```
Comment (xutian) [inline virttest/utils_net.py]: please use lowercase in function name.

File: virttest/utils_net.py [hunk 4]
```diff
@@ -3936,19 +3936,19 @@ def _check_file_size_unchanged():
...
     :param vm: target vm
-    :param timeout: the timeout of login
-    :return: a tuple of (msis, queues)
+    :param timeout: timeout value for login
+    :return: the content of traceview log file
     """
-
     session = vm.wait_for_login(timeout=timeout)
     # prepare traceview environment
     pdb_local_path, traceview_local_path = _prepare_traceview_windows(params,
```
Comment (xutian) [inline virttest/utils_net.py]: local unused var 'vm' in this function.

File: virttest/utils_net.py [hunk 5]
```diff
@@ -3997,4 +3997,20 @@ def get_MSIs_and_queues_windows(params, vm, timeout=360):
...
+    """
+    Get MSIs&queues infomation of current windows guest.
+    First start a traceview session, then restart the nic interface
+    to trigger logging. By analyzing the dumped output, the MSIs&queues
+    info is acquired.
+
+    :param params: the test params
+    :param vm: target vm
+    :param timeout: the timeout of login
+    :return: a tuple of (msis, queues)
```
Comment (xutian) [inline virttest/utils_net.py]: MIS(s) in  doc string

File: virttest/utils_net.py [hunk 6]
```diff
@@ -3997,4 +3997,20 @@ def get_MSIs_and_queues_windows(params, vm, timeout=360):
         raise exceptions.TestError("Cann't read dumped file %s: %s" %
                                    (dump_file, output))
     session_serial.close()
+    return output
+
+
+def get_MSIs_and_queues_windows(params, vm, timeout=360):
+    """
+    Get MSIs&queues infomation of current windows guest.
```
Comment (xutian) [inline virttest/utils_net.py]: Get MSI and queue's  information from VM windows OS.

File: virttest/utils_net.py [hunk 7]
```diff
@@ -3997,4 +3997,20 @@ def get_MSIs_and_queues_windows(params, vm, timeout=360):
         raise exceptions.TestError("Cann't read dumped file %s: %s" %
                                    (dump_file, output))
     session_serial.close()
```
Comment (xutian) [inline virttest/utils_net.py]: close serial session in finally section to avoid "Device in used error"

**Comment by kimi1978wy** [comment]
(1/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.mq_enabled_chk.q35: PASS (355.69 s)
 (2/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.boot_with_max_queues.q35: PASS (382.79 s)
 (3/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.netkvm_cpu_mapping.q35: PASS (158.91 s)

File: virttest/utils_net.py [hunk 8]
```diff
@@ -3936,65 +3935,82 @@ def _check_file_size_unchanged():
...
-    if status:
-        logging.error("Cann't dump log file %s: %s" % (log_path, output))
-    _wait_for_traceview_dump_finished(session_serial, dump_file)
-    status, output = session_serial.cmd_status_output("type %s" % dump_file)
-    if status:
-        raise exceptions.TestError("Cann't read dumped file %s: %s" %
-                                   (dump_file, output))
-    session_serial.close()
-    return _get_MSI_queue_from_traceview_output(output)
+    try:
```
Comment (xutian) [inline virttest/utils_net.py]:
```
    def dump_traceview_log_windows(params, vm, timeout=360):
    """
    Dump traceview log file with nic restart panic
    Steps:
        1.Prepare traceview.exe & driver pdb files
        2.Start traceview session
        2.Restart network card to create panic
        3.Stop traceview session and dump log file

    :param params: test params
    :param vm: target vm
    :param timeout: timeout value for login
    :return: the content of traceview log file
    """
    log_path = "c:\\logfile.etl"
    clean_cmd = "del " + log_path
    dump_file = "c:\\trace.txt"

    session = vm.wait_for_login(timeout=timeout)
    # prepare traceview environment
    pdb_local_path, traceview_local_path = _prepare_traceview_windows(
        params, session, timeout)
    session.close()
    start_traceview_cmd = "%s -start test_session -pdb %s -level 5 -flag 0x1fff -f %s" % (
        traceview_local_path, pdb_local_path, log_path)
    stop_traceview_cmd = "%s -stop test_session" % traceview_local_path
    dump_cmd = "%s -process %s -pdb %s -o %s" % (
        traceview_local_path, log_path, pdb_local_path, dump_file)
    # start traceview
    logging.info("Start trace view with pdb file")
    session_serial = vm.wait_for_serial_login(timeout=timeout)
    try:
        session_serial.cmd(clean_cmd)
        session_serial.cmd(start_traceview_cmd, timeout=timeout)
        # restart nic
        logging.info("Restart guest nic")
        mac = vm.get_mac_address(0)
        connection_id = get_windows_nic_attribute(
            session_serial, "macaddress", mac, "netconnectionid")
        restart_windows_guest_network(session_serial, connection_id)
        # stop traceview
        logging.info("Stop traceview")
        session_serial.cmd(stop_traceview_cmd, timeout=timeout)
        # checkout traceview output
        logging.info("Check etl file generated by traceview")
        session_serial.cmd(clean_cmd)
        status, output = session_serial.cmd_status_output(dump_cmd)
        if status:
            logging.error("Cann't dump log file %s: %s" % (log_path, output))
        _wait_for_traceview_dump_finished(session_serial, dump_file)
        status, output = session_serial.cmd_status_output(
            "type %s" % dump_file)
        if status:
            raise exceptions.TestError(
                "Cann't read dumped file %s: %s" % (dump_file, output))
        return output
    finally:
        session_serial.close()
```

Comment (xutian) [reply virttest/utils_net.py]: reformat your function, please have a test.

**Comment by kimi1978wy** [comment]
(1/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.netkvm_cpu_mapping.q35: PASS (160.06 s)
 (2/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.mq_enabled_chk.q35: PASS (358.28 s)
 (3/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.boot_with_max_queues.q35: PASS (403.11 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Comment by kimi1978wy** [comment]
@xutian updated, you can review it again, thanks a lot

**Pull request merged**